index.bs

```diff
@@ -5638,6 +5639,97 @@ At this time, one [=credential property=] is defined: the [=resident key credent
+           1. Set `hmac-secret` to [TRUE] in the authenticator extensions input.
+           1. If {{AuthenticationExtensionsPRFInputs/eval}} is present and a future extension to [[FIDO-CTAP]] permits evaluation of the PRF at creation time, configure `hmac-secret` inputs accordingly:
+               * Let `salt1` be the value of <code>SHA-256(UTF8Encode("WebAuthn PRF") || 0x00 || {{AuthenticationExtensionsPRFValues/first}})</code>.
+               * If {{AuthenticationExtensionsPRFValues/second}} is present, let `salt2` be the value of <code>SHA-256(UTF8Encode("WebAuthn PRF") || 0x00 || {{AuthenticationExtensionsPRFValues/second}})</code>.
+           1. Set {{AuthenticationExtensionsPRFOutputs/enabled}} to the value of `hmac-secret` in the authenticator extensions output. If not present, set {{AuthenticationExtensionsPRFOutputs/enabled}} to [FALSE].
+           1. Set {{AuthenticationExtensionsPRFOutputs/results}} to the PRF outputs, if any.
+
+: Client extension output ([=registration extension|registration=])
+:: <xmp class="idl">
+    dictionary AuthenticationExtensionsPRFOutputs {
+        boolean enabled;
+        AuthenticationExtensionsPRFValues results;
+    };
+
+    partial dictionary AuthenticationExtensionsClientOutputs {
+        AuthenticationExtensionsPRFOutputs prf;
+    };
+    </xmp>
+
+    <div dfn-type="dict-member" dfn-for="AuthenticationExtensionsPRFOutputs">
+        :   <dfn>enabled</dfn>
+        ::  [TRUE] if, and only if, the one or two PRFs are available for use with the created credential.
+
+        :   <dfn>results</dfn>
+        ::  The results of evaluating the PRF for the inputs given in {{AuthenticationExtensionsPRFInputs/eval}} or {{AuthenticationExtensionsPRFInputs/evalByCredential}}. Outputs may not be available during [=registration=]; see comments in {{AuthenticationExtensionsPRFInputs/eval}}.
+    </div>
+
+: Client extension processing ([=authentication extension|authentication=])
+::
+       1. If {{AuthenticationExtensionsPRFInputs/evalByCredential}} is present and {{PublicKeyCredentialRequestOptions/allowCredentials}} is empty, return a {{DOMException}} whose name is “{{NotSupportedError}}”.
+       1. If either {{AuthenticationExtensionsPRFInputs/evalByCredential}} is present or {{AuthenticationExtensionsPRFInputs/eval}} is present, and {{PublicKeyCredentialRequestOptions/userVerification}} has the value {{UserVerificationRequirement/preferred}}, return a {{DOMException}} whose name is “{{NotSupportedError}}”.
+       1. Find any applicable PRF input(s):
+           1. If {{AuthenticationExtensionsPRFInputs/evalByCredential}} is present and contains an entry whose key is the [=base64url encoding=] of the [=credential ID=] that will be returned, use the value of that entry.
+           1. If no input(s) were found in the previous step, and if {{AuthenticationExtensionsPRFInputs/eval}} is present, then use that value.
+       1. If PRF input(s) were found:
```

Comment:
I think we should explicitly define what should happen if no inputs were found. I read this to mean the client should then return an empty map as the extension output, but I imagine it could also be interpreted as to omit the extension output if it would be empty.

Reply:
Good point, thanks. I've altered things to be explicit that not finding an input gives `prf: {}` in the output extensions. (I.e. the RP can tell the difference between the PRF extension not being supported, and a suitable input not being found.)